Vagrantfile

```diff
@@ -1,71 +1,320 @@
+        "2.1-6"
+      else
+        "2.0-5"
+      end
+
+    c.vm.provision :shell, inline: %(
+      echo '' > /etc/profile.d/crystal.sh
+
+      apt-get install -y apt-transport-https curl build-essential git
+      apt-get install -y libxml2-dev libyaml-dev libreadline-dev libgmp3-dev libssl-dev libpcre3-dev \
+        libevent-dev libevent-core-#{libevent_ver} libevent-extra-#{libevent_ver} libevent-openssl-#{libevent_ver} libevent-pthreads-#{libevent_ver}
+
+      #{install_llvm(dist, llvm)}
+
+      #{install_cxx}
+    )
+
+    install_crystal(c, "ubuntu", dist: dist, bits: bits, github_targz: github_targz)
+
+    clone_crystal_from_vagrant(c)
+  end
+end
+
+# define a ubuntu box with the given *name*
+# *llvm* values: 6.0, 7, (empty), Hash with {url:, path:}
+def define_debian(config, name:, dist:, bits:, llvm: nil, github_targz: INSTALL_GITHUB_TARGZ)
+  config.vm.define(name) do |c|
+    c.vm.box = "debian/#{dist}#{bits}"
+
+    setup_memory(config, bits)
+
+    c.vm.provision :shell, inline: %(
+      echo '' > /etc/profile.d/crystal.sh
+
+      apt-get install -y software-properties-common apt-transport-https dirmngr curl build-essential git pkg-config
+      apt-get install -y zlib1g-dev libxml2-dev libyaml-dev libreadline-dev libgmp3-dev libssl-dev libpcre3-dev \
+        libevent-dev libevent-core-2.0-5 libevent-extra-2.0-5 libevent-openssl-2.0-5 libevent-pthreads-2.0-5
+
+      #{install_llvm(dist, llvm)}
+    )
+
+    install_crystal(c, "debian", dist: dist, bits: bits, github_targz: github_targz)
+
+    clone_crystal_from_vagrant(c)
   end
+end
+
+def define_freebsd(config, name:, box:)
+  config.vm.define name do |c|
+    c.ssh.shell = "sh"
+    c.vm.box = "freebsd/FreeBSD-#{box}"
+    setup_memory(config, 64)
 
-  config.vm.define "freebsd" do |c|
-    c.ssh.shell = "csh"
-    c.vm.box = "freebsd/FreeBSD-10.2-RELEASE"
+    c.vm.hostname = name
+    c.vm.base_mac = "6658695E16F0"
 
     c.vm.network "private_network", type: "dhcp"
     c.vm.synced_folder ".", "/vagrant", type: "nfs"
 
-    # to build boehm-gc from git repository:
-    #c.vm.provision :shell, inline: %(
-    #  pkg install -y libtool automake autoconf libatomic_ops
-    #)
-
     c.vm.provision :shell, inline: %(
-      pkg install -y git gmake pkgconf pcre libunwind clang36 libyaml gmp libevent2
+      mkdir -p /etc/profile.d
+      echo '' > /etc/profile.d/crystal.sh
+      echo '. /etc/profile.d/crystal.sh' >> /etc/profile
+
+      pkg install -y git gmake pkgconf
+      pkg install -y libyaml gmp libunwind libevent pcre boehm-gc-threaded
+      pkg install -y llvm60
     )
 
-    clone_crystal_from_vagrant.call(c)
+    install_crystal(c, "freebsd", dist: nil, bits: 64, github_targz: false)
+
+    clone_crystal_from_vagrant(c)
   end
+end
 
-  config.vm.provider "virtualbox" do |vb|
-    vb.memory = 4096
-    vb.cpus = 2
+def define_alpine(config, name:, bits:, llvm: '4')
+  config.vm.define name do |c|
+    c.vm.box = "alpine/alpine#{bits}"
```

Comment:
[`alpine/alpine64`](https://app.vagrantup.com/alpine/boxes/alpine64) doesn't seem to be a great vagrant box. It's current version is still 3.7.0 which was released in 2017.
The most-used alpine box is [`generic/alpine38`](https://app.vagrantup.com/generic/boxes/alpine38) (`alpine39` is also available).